### PR TITLE
Add example for AudioWorkletNode constructor options parameter

### PR DESCRIPTION
**Pull request by DivyamAhuja**
I wrote this example for issue regarding #131 .
In this example we can pass type and frequency options to AudioWorkletNode constructor and use those in AudioWorkletProcessor.

Here is how it looks like.
https://divyamahuja.github.io/web-audio-samples/audio-worklet/basic/audio-worklet-node-options/ 

I wonder if this will work for #131 issue

### CODE REVIEW

File: audio-worklet/basic/audio-worklet-node-options/PageData.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  TopBar: {
+    pathData: [
+      ['Home', '../../../'],
+      ['AudioWorklet', '../../'],
+      ['AudioWorkletNode Options!'],
+    ],
+  },
+
+  Description: {
+    title: 'AudioWorkletNode Options!!',
```
Comment (hoch) [inline audio-worklet/basic/audio-worklet-node-options/PageData.js:26]: Remove `!!`.

File: audio-worklet/basic/audio-worklet-node-options/PageData.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+/**
+ * Copyright 2018 Google Inc. All Rights Reserved.
```
Comment (hoch) [inline audio-worklet/basic/audio-worklet-node-options/PageData.js:2]: It should be 2022.

File: audio-worklet/PageData.js [hunk 3]
```diff
@@ -52,6 +52,12 @@ export default {
         description: `Messaging between node and processor`,
         url: 'basic/message-port/',
       },
+      {
+        title: 'AudioWorkletNode Options!',
```
Comment (hoch) [inline audio-worklet/PageData.js:56]: Remove `!`

File: audio-worklet/basic/audio-worklet-node-options/basic-processor.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
+
+// https://en.wikipedia.org/wiki/Sawtooth_wave
+const sawtooth = (f, t) => {
```
Comment (hoch) [inline audio-worklet/basic/audio-worklet-node-options/basic-processor.js:3]: No abbreviation on arguments:
https://google.github.io/styleguide/jsguide.html#naming-rules-common-to-all-identifiers

Comment (hoch) [reply audio-worklet/basic/audio-worklet-node-options/basic-processor.js:3]: Ditto for other functions below.

File: audio-worklet/basic/audio-worklet-node-options/basic-processor.js [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+// https://en.wikipedia.org/wiki/Triangle_wave
+const triangle = (f, t) => {
+  return 2 * Math.abs(2 * (f * t - Math.floor(f * t + 0.5))) - 1
+}
+
+const noise = () => {
+  return 2 * (Math.random() - 0.5);
+}
+
+class BasicProcessor extends AudioWorkletProcessor {
```
Comment (hoch) [inline audio-worklet/basic/audio-worklet-node-options/basic-processor.js:26]: Add an overview comment for the class definition:
https://google.github.io/styleguide/jsguide.html#jsdoc-class-comments

File: audio-worklet/basic/audio-worklet-node-options/basic-processor.js [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+  constructor(options) {
+      super();
+      this.sample = 0;
+      this.output_function = noise;
+      this.f = 440;
```
Comment (hoch) [inline audio-worklet/basic/audio-worklet-node-options/basic-processor.js:31]: No abbreviation.

File: audio-worklet/basic/audio-worklet-node-options/basic-processor.js [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+  return 2 * Math.abs(2 * (f * t - Math.floor(f * t + 0.5))) - 1
+}
+
+const noise = () => {
+  return 2 * (Math.random() - 0.5);
+}
+
+class BasicProcessor extends AudioWorkletProcessor {
+  constructor(options) {
+      super();
```
Comment (hoch) [inline audio-worklet/basic/audio-worklet-node-options/basic-processor.js:28]: 2-space indentation:
https://google.github.io/styleguide/jsguide.html#formatting-block-indentation

File: audio-worklet/basic/audio-worklet-node-options/basic-processor.js [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+              this.output_function = noise;
+          }
+          this.f = options.processorOptions.frequency || 440;
+      }
+  }
+
+  process(inputs, outputs, parameters) {
+
+      const output = outputs[0];
+
```
Comment (hoch) [inline audio-worklet/basic/audio-worklet-node-options/basic-processor.js:51]: Redundant vertical whitespaces:
https://google.github.io/styleguide/jsguide.html#formatting-vertical-whitespace

File: audio-worklet/basic/audio-worklet-node-options/index.html [hunk 9]
```diff
@@ -0,0 +1,106 @@
+<!--
+ * Copyright 2018 Google Inc. All Rights Reserved.
```
Comment (hoch) [inline audio-worklet/basic/audio-worklet-node-options/index.html:2]: 2018 --> 2022

File: audio-worklet/basic/audio-worklet-node-options/index.html [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+    import PageData from './PageData.js';
+
+    import {
+      html, render,
+      Component, WorkletIndicator, TopBar, Footer,
+      OneColumnView
+    }
+      from '../../../assets/Components.js';
+
+
```
Comment (hoch) [inline audio-worklet/basic/audio-worklet-node-options/index.html:39]: Redundant vertical whitespaces:
https://google.github.io/styleguide/jsguide.html#formatting-vertical-whitespace

File: audio-worklet/basic/audio-worklet-node-options/index.html [hunk 11]
```diff
@@ -0,0 +1,106 @@
...
+        <label for="type">Type:</label>
+        <select name="type" id="demo-input-type">
+          <option value="noise">Noise</option>
+          <option value="sine">Sine</option>
+          <option value="square">Square</option>
+          <option value="triangle">Triangle</option>
+          <option value="sawtooth">Sawtooth</option>
+        </select>
+        <label for="frequency">Frequency (1 - 20000):</label>
+        <input type="number" name="frequency" id="demo-input-frequency" min="1" max="20000" value=440>
```
Comment (hoch) [inline audio-worklet/basic/audio-worklet-node-options/index.html:54]: Didn't count, but we wrap the code at column 80:
https://google.github.io/styleguide/jsguide.html#formatting-column-limit

File: audio-worklet/basic/audio-worklet-node-options/index.html [hunk 12]
```diff
@@ -0,0 +1,106 @@
...
+          context.suspend()
+          context.close();
+        }
+      }
+    }
+
+    document.querySelector('#toogle').addEventListener('click', demoFunction);
+    document.querySelector('#demo-input-type').addEventListener('change', demoFunction);
+    document.querySelector('#demo-input-frequency').addEventListener('change', demoFunction);
+
```
Comment (hoch) [inline audio-worklet/basic/audio-worklet-node-options/index.html:100]: Redundant vertical whitespaces.

File: audio-worklet/basic/audio-worklet-node-options/basic-processor.js [hunk 13]
```diff
@@ -1,65 +1,73 @@
...
-              this.output_function = square;
-          } else if (options.processorOptions.type == 'triangle') {
-              this.output_function = triangle;
-          } else if (options.processorOptions.type == 'noise') {
-              this.output_function = noise;
-          }
-          this.f = options.processorOptions.frequency || 440;
+    super();
+    this.sample = 0;
+    this.output_function = noise;
```
Comment (hoch) [inline audio-worklet/basic/audio-worklet-node-options/basic-processor.js:40]: `outputFunction`

File: audio-worklet/basic/audio-worklet-node-options/index.html [hunk 14]
```diff
@@ -93,13 +93,12 @@
...
     }
-
+    
     document.querySelector('#toogle').addEventListener('click', demoFunction);
-    document.querySelector('#demo-input-type').addEventListener('change', demoFunction);
-    document.querySelector('#demo-input-frequency').addEventListener('change', demoFunction);
-
-
-
+    document.querySelector('#demo-input-type').addEventListener('change',
```
Comment (hoch) [inline audio-worklet/basic/audio-worklet-node-options/index.html:98]: https://google.github.io/styleguide/jsguide.html#formatting-indent

I would actually prefer:
```js
document.querySelector('#demo-input-type')
        .addEventListener('change', demoFunction);
```

**Review by hoch — COMMENTED** [review]
We're almost there. Just a few nits.

File: audio-worklet/basic/audio-worklet-node-options/index.html (hunk 14 above)
Comment (DivyamAhuja) [reply audio-worklet/basic/audio-worklet-node-options/index.html:98]: Sure!! That looks better

File: audio-worklet/basic/audio-worklet-node-options/basic-processor.js (hunk 13 above)
Comment (hoch) [reply audio-worklet/basic/audio-worklet-node-options/basic-processor.js:42]: I guess this is not fixed yet?

Comment (DivyamAhuja) [reply audio-worklet/basic/audio-worklet-node-options/basic-processor.js]: Sorry, I almost missed that XD

File: audio-worklet/basic/audio-worklet-node-options/basic-processor.js [hunk 15]
```diff
@@ -0,0 +1,76 @@
...
+    }
+  }
+
+  process(inputs, outputs, parameters) {
+    const output = outputs[0];
+
+    for (let channel = 0; channel < output.length; ++channel) {
+      const outputChannel = output[channel];
+      for (let i = 0; i < outputChannel.length; ++i) {
+        outputChannel[i] = this.outputFunction(this.frequency,
```
Comment (hoch) [inline audio-worklet/basic/audio-worklet-node-options/basic-processor.js:66]: Perhaps:

```js
outputChannel[i] =
    this.outputFunction(this.frequency, this.sample / sampleRate);
```

File: audio-worklet/basic/audio-worklet-node-options/index.html [hunk 16]
```diff
@@ -0,0 +1,105 @@
...
+            type: type,
+            frequency: frequency
+          }
+        });
+        basicProcessor.connect(context.destination);
+        context.resume();
+      } else {
+        element.innerText = 'START';
+        if (context) {
+          context.suspend()
```
Comment (hoch) [inline audio-worklet/basic/audio-worklet-node-options/index.html:91]: A missing semicolon.

File: audio-worklet/basic/audio-worklet-node-options/index.html [hunk 17]
```diff
@@ -0,0 +1,105 @@
...
+  <div class="container was-page-wrap"></div>
+  <script type="module">
+    import PageData from './PageData.js';
+
+    import {
+      html, render,
+      Component, WorkletIndicator, TopBar, Footer,
+      OneColumnView
+    }
+      from '../../../assets/Components.js';
```
Comment (hoch) [inline audio-worklet/basic/audio-worklet-node-options/index.html:37]: We can move this up to L.36 after the curly brace.

**Review by hoch — APPROVED** [review]
LGTM with nits, mostly coding styles.

Thanks for staying on top of this!

**Review by hoch — APPROVED** [review]

**Pull request merged**